gsitemap.php

```diff
@@ -1,907 +1,987 @@
+                die();
+            } else {
+                if ($this->cron) {
+                    header('location: http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true).__PS_BASE_URI__.'modules/gsitemap/gsitemap-cron.php?continue=1&token='.Tools::substr(Tools::encrypt('gsitemap/cron'), 0, 10).'&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
+                } else {
+                    $admin_folder = str_replace(_PS_ROOT_DIR_, '', _PS_ADMIN_DIR_);
+                    $admin_folder = Tools::substr($admin_folder, 1);
+                    header('location: http'.(Configuration::get('PS_SSL_ENABLED') ? 's' : '').'://'.Tools::getShopDomain(false, true).__PS_BASE_URI__.$admin_folder.'/index.php?tab=AdminModules&configure=gsitemap&token='.Tools::getAdminTokenLite('AdminModules').'&tab_module='.$this->tab.'&module_name=gsitemap&continue=1&type='.$new_link['type'].'&lang='.$lang.'&index='.$index.'&id='.(int)$id_obj.'&id_shop='.$this->context->shop->id);
+                }
+                die();
+            }
+        }
+    }
+
+    /**
+     * Hydrate $link_sitemap with home link
+     *
+     * @param array  $link_sitemap contain all the links for the Google Sitemap file to be generated
+     * @param string $lang         language of link to add
+     * @param int    $index        index of the current Google Sitemap file
+     * @param int    $i            count of elements added to sitemap main array
+     *
+     * @return bool
+     */
+    protected function _getHomeLink(&$link_sitemap, $lang, &$index, &$i)
+    {
+        $link = new Link();
+        return $this->_addLinkToSitemap(
+            $link_sitemap,
+            array(
+                'type' => 'home',
+                'page' => 'home',
+                'link' => $link->getPageLink('index', null, $lang['id_lang']),
+                'image' => false
+            ),
+            $lang['iso_code'],
+            $index,
+            $i,
+            -1
+        );
+    }
+
+    /**
+     * Hydrate $link_sitemap with meta link
+     *
+     * @param array  $link_sitemap contain all the links for the Google Sitemap file to be generated
+     * @param string $lang         language of link to add
+     * @param int    $index        index of the current Google Sitemap file
+     * @param int    $i            count of elements added to sitemap main array
+     * @param int    $id_meta      meta object identifier
+     *
+     * @return bool
+     */
+    protected function _getMetaLink(&$link_sitemap, $lang, &$index, &$i, $id_meta = 0)
+    {
+        if (method_exists('ShopUrl', 'resetMainDomainCache')) {
+            ShopUrl::resetMainDomainCache();
+        }
+        $link = new Link();
+        if (version_compare(_PS_VERSION_, '1.6', '>=')) {
```

Comment:
This if shouldn't be needed anymore if you up the minimum compatibility to 1.7+

Reply:
Fixed